src/index.ts

```diff
@@ -0,0 +1,213 @@
+import { CancelButton } from "./elements/CancelButton";
+import { DeleteButton } from "./elements/DeleteButton";
+import { SaveButton } from "./elements/SaveButton";
+import { Annotation } from "./types/Annotation";
+
+/**
+ * Custom annotation editor for Geniza project
+ */
+class TranscriptionEditor {
+    anno;
+
+    annotationContainer: HTMLElement;
+
+    storage;
+
+    // TODO: Add typedefs for the Annotorious client (anno) and storage plugin
+
+    /**
+     * Initialize an instance of the TranscriptionEditor.
+     *
+     * @param {any} anno Instance of Annotorious.
+     * @param {any} storage Storage plugin to save Annotorious annotations.
+     * @param {HTMLElement} annotationContainer Existing HTML element that the editor will be
+     * placed into.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    constructor(anno: any, storage: any, annotationContainer: HTMLElement) {
+        this.anno = anno;
+        this.storage = storage;
+        // disable the default annotorious editor (headless mode)
+        this.anno.disableEditor = true;
+        this.annotationContainer = annotationContainer;
+
+        // define custom elements
+        customElements.define("save-button", SaveButton, { extends: "button" });
+        customElements.define("cancel-button", CancelButton, {
+            extends: "button",
+        });
+        customElements.define("delete-button", DeleteButton, {
+            extends: "button",
+        });
+        customElements.define("annotation-block", AnnotationBlock, {
+            extends: "div",
+        });
+
+        // attach event listeners
+        document.addEventListener(
+            "annotations-loaded",
+            this.handleAnnotationsLoaded.bind(this),
+        );
+        this.anno.on("createSelection", this.handleCreateSelection.bind(this));
+        this.anno.on(
+            "selectAnnotation",
+            this.handleSelectAnnotation.bind(this),
+        );
+    }
+
+    /**
+     * Handler for custom annotations loaded event triggered by storage plugin.
+     */
+    handleAnnotationsLoaded() {
+        // remove any existing annotation displays, in case of update
+        this.annotationContainer
+            .querySelectorAll(".annotation-display-container")
+            .forEach((el) => el.remove());
+        // display all current annotations
+        this.anno.getAnnotations().forEach((annotation: Annotation) => {
+            this.annotationContainer.append(
+                new AnnotationBlock({
+                    annotation,
+                    editable: false,
+                    onCancel: this.anno.cancelSelected,
+                    onClick: this.handleClickAnnotationBlock.bind(this),
+                    onDelete: this.handleDeleteAnnotation.bind(this),
+                    onSave: this.handleSaveAnnotation.bind(this),
+                    updateAnnotorious: this.anno.addAnnotation,
```

Comment:
I agree with you about disliking how we're binding things currently. 

Can we resolve this by approaching the way we're supposed to for Turbo anyway, by using event delegation and binding once on document or window? Discussed briefly here https://turbo.hotwired.dev/handbook/building#observing-navigation-events 

Let me know if this isn't clear. I'm not sure we need to address now, unless it solves a problem.

Reply:
That makes sense to me—put the event listeners on the entire transcription editor and not just each block. I agree it's not necessary now, more of a cleanup chore.

Reply:
FWIW, these events are being attached after the `document` event `annotations-loaded` is fired, so they're meeting the Turbo expectation to some extent, but we are breaking it in other places.